documentation/FortranForCProgrammers.md

```diff
@@ -0,0 +1,254 @@
+in parentheses.
+
+The compiler is free to assume that any aliasing between dummy arguments
+and other data is safe.
+In other words, if some object can be written to under one name, it's
+never going to be read or written using some other name in that same
+scope.
+```
+  SUBROUTINE FOO(X,Y,Z)
+  X = 3.14159
+  Y = 2.1828
+  Z = 2 * X ! CAN BE FOLDED AT COMPILE TIME
+  END
+```
+This is the opposite of the assumptions under which a C or C++ compiler must
+labor when trying to optimize code with pointers.
+
+Polymorphism
+------------
+Fortran code can be written to accept data of some derived type or
+any extension thereof using `CLASS`, deferring the actual type to
+execution, rather than the usual `TYPE` syntax.
+This is somewhat similar to the use of `virtual` functions in c++.
+
+Fortran's `SELECT TYPE` construct is used to distinguish between
+possible specific types dynamically.
+
+Pointers
+--------
+Pointers are objects in Fortran, not a data type.
+Pointers can point to data, arrays, and subprograms, but not to other pointers
+or to an allocatable.
+A pointer can only point to data that has the `TARGET` attribute.
+Outside of the pointer assignment statement (`P=>X`) and some intrinsic
+functions and cases with pointer dummy arguments, pointers are implicitly
+dereferenced, and the use of their name is a reference to the data to which
+they point instead.
+
+Unlike allocatables, pointers do not deallocate their data when they go
+out of scope.
+
+A legacy feature, "Cray pointers", implements dynamic base addressing of
+one variable using an address stored in another.
+
+Preprocessing
+-------------
+There is no standard preprocessing feature, but every real Fortran implementation
+has some support for passing Fortran source code through a variant of
+the standard C source preprocessor.
+Behavior varies across implementations and one should not depend on
+much portability.
+Preprocessing is typically requested by the use of a capitalized filename
+suffix (e.g., "foo.F90") or a compiler command line option.
+
+Pitfalls
+--------
+Variable initializers, e.g. `INTEGER :: J=123`, are _static_ initializers!
+They imply that the variable is stored in static storage, not on the stack,
+and the initialized value lasts only until the variable is assigned.
+One must use an assignment statement to implement a dynamic initializer
+that will apply to every fresh instance of the variable.
+(Derived type component initializers, however, do work as expected.)
+
+If you see an assignment to an array that's never been declared as such,
+it's probably a definition of a "statement function", which is like
+a parameterized macro definition, e.g. "A(X)=SQRT(X)**3".
+In the original Fortran language, this was the only means for user
+function definitions.
+Today, of course, one should use an external or internal function instead.
+
+Fortran expressions don't bind exactly like C's do.
+Watch out for exponentiation with `**`, which of course C lacks; it
```

Comment:
And this ended the bolding.  Note that putting ** in code blocks does not cause bolding.  `**1`